extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcAuthenticationMechanism.java

```diff
@@ -42,20 +44,28 @@ public OidcAuthenticationMechanism(DefaultTenantConfigResolver resolver, Blockin
     }
 
     @Override
-    public Uni<SecurityIdentity> authenticate(RoutingContext context,
-            IdentityProviderManager identityProviderManager) {
+    public Uni<SecurityIdentity> authenticate(RoutingContext context, IdentityProviderManager identityProviderManager) {
         return resolve(context).chain(new Function<>() {
             @Override
             public Uni<? extends SecurityIdentity> apply(OidcTenantConfig oidcConfig) {
                 if (!oidcConfig.tenantEnabled()) {
                     return Uni.createFrom().nullItem();
                 }
-                return isWebApp(context, oidcConfig) ? codeAuth.authenticate(context, identityProviderManager, oidcConfig)
-                        : bearerAuth.authenticate(context, identityProviderManager, oidcConfig);
+                OidcAuthenticationPolicy authenticationPolicy = context.get(AUTHENTICATION_POLICY_KEY);
```

Comment:
I'm not sure about this. It should all be constrained to the interceptor which does this check

Reply:
> I'm not sure about this. It should all be constrained to the interceptor which does this check

The interceptor runs before authentication, not after it (which is too late, this is part of authentication because it can fail authentication). I can move this check around, this is definitely not the only way....